core/shared/src/main/scala/io/udash/properties/DefaultValue.scala

```diff
@@ -0,0 +1,87 @@
+package io.udash.properties
+
+import com.avsystem.commons.misc.{AbstractCase, Opt}
+
+import scala.annotation.implicitNotFound
+import scala.collection.generic.CanBuildFrom
+import scala.language.higherKinds
+
+@implicitNotFound(
+  "Class ${A} does not have a default value. Please, specify the value of this property or add `implicit val default: DefaultValue[${A}] = ???` in ${A}'s companion object."
+)
+trait DefaultValue[A] {
+  def value: A
+}
+
+object DefaultValue extends LowPrioImplicits {
+  case class Simple[A](value: A) extends AbstractCase with DefaultValue[A]
+  case class Factory[A](factory: () => A) extends AbstractCase with DefaultValue[A] {
+    override def value: A = factory()
+  }
+
+  implicit val double: DefaultValue[Double] = Simple(0.0)
+  implicit val float: DefaultValue[Float] = Simple(0.0f)
+  implicit val long: DefaultValue[Long] = Simple(0L)
+  implicit val int: DefaultValue[Int] = Simple(0)
+  implicit val short: DefaultValue[Short] = Simple(0)
+  implicit val byte: DefaultValue[Byte] = Simple(0)
+  implicit val char: DefaultValue[Char] = Simple(' ')
+  implicit val boolean: DefaultValue[Boolean] = Simple(false)
+  implicit val unit: DefaultValue[Unit] = Simple(())
+  implicit val string: DefaultValue[String] = Simple("")
+
+  implicit def option[A]: DefaultValue[Option[A]] = Simple(None)
+  implicit def opt[A]: DefaultValue[Opt[A]] = Simple(Opt.Empty)
```

Comment:
These two will always create an instance of `Simple`. Not necessary, imho. Use one instance and cast, maybe?